### DISCUSSION TITLE
Any description of used lns variants ?

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#3581] [CP-SAT questions] [answered]
Hello,
  I was not satisfied with performance so I tried to simply fix order of intervals for every resource and found that I can get better results much faster. Optimization minimize makespan. 
  
  Althought I know that cp-sat doesn’t support custom lns, this seems to me as too general to be not used. Maybe only definition of resource for which order of intervals should be preserved can be tricky.

  So I want to know whether such lns variant exists, and maybe why it doesn’t work in my case.

Jan

**Comment by lperron** [comment, marked as answer]
There are all described in https://github.com/google/or-tools/blob/stable/ortools/sat/cp_model_lns.h
